leptos_macro/src/component.rs

```diff
@@ -655,14 +655,43 @@ impl Prop {
                 abort!(e.span(), e.to_string());
             });
 
-        let name = if let Pat::Ident(i) = *typed.pat {
-            i
-        } else {
-            abort!(
-                typed.pat,
-                "only `prop: bool` style types are allowed within the \
-                 `#[component]` macro"
-            );
+        let name = match *typed.pat {
+            Pat::Ident(i) => {
+                if let Some(name) = &prop_opts.name {
+                    PatIdent {
```

Comment:
This also allows to use the new `name` attribute on props which are not destructured.   
That means props can have different names inside the component and outside (builder).

We can also error here if we don't want to allow renaming normal props